core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -7,21 +7,36 @@ package akka.kafka
 import java.util
 import java.util.concurrent.atomic.AtomicInteger
 import java.util.regex.Pattern
+
 import akka.actor.{Actor, ActorLogging, ActorRef, Cancellable, Props, Status, Terminated}
 import akka.event.LoggingReceive
 import org.apache.kafka.clients.consumer._
 import org.apache.kafka.common.TopicPartition
 import org.apache.kafka.common.errors.WakeupException
+
 import scala.collection.JavaConverters._
 import java.util.concurrent.locks.LockSupport
+
+import akka.Done
 import akka.actor.DeadLetterSuppression
+
 import scala.util.control.{NoStackTrace, NonFatal}
 import scala.util.{Failure, Success, Try}
 
 object KafkaConsumerActor {
   case class StoppingException() extends RuntimeException("Kafka consumer is stopping")
-  def props[K, V](settings: ConsumerSettings[K, V]): Props =
-    Props(new KafkaConsumerActor(settings)).withDispatcher(settings.dispatcher)
+
+  def propsWithCommitResult[K, V](
+    settings: ConsumerSettings[K, V]
+  ): Props = {
+    Props(new KafkaConsumerActor(settings, m => Internal.OffsetCommitResult(m.asScala.toMap))).withDispatcher(settings.dispatcher)
```

Comment:
I don't understand this, what is the purpose of this `OffsetCommitResult`? I don't see it used anywhere.